src/veritysetup/veritysetup.c

```diff
@@ -43,6 +46,59 @@ static int help(void) {
         return 0;
 }
 
+static int parse_options(const char *options) {
+        _cleanup_free_ const char *copy = NULL;
+        int r;
+
+        copy = strdup(options);
+        if (!copy)
+                return log_oom();
+
+        for (;;) {
+                _cleanup_free_ char *word = NULL;
+                struct stat st;
+                char *val;
+
+                r = extract_first_word(&copy, &word, ",", EXTRACT_DONT_COALESCE_SEPARATORS | EXTRACT_UNESCAPE_SEPARATORS);
```

Comment:
This will leak the dupped memory of "copy" (and result in memory corruption later) because the copy ptr is advanced and thus the the `_cleanup_free_` above destroy a pointer that points to the end of the actually allocated block

Reply:
That was a misunderstanding of https://github.com/systemd/systemd/pull/17576#discussion_r540191192